### ISSUE TITLE
intended use of integrate_blindly

### ISSUE THREAD

**Issue by LukeLabrie** [neurophysik/jitcdde#57]
This is a usage question, but also maybe a potential issue. I noticed in the examples that when using `integrate`, generating a solution has the form

```python
# Define system...

sol = []
for t_x in T:
    sol.append(DDE.integrate(t_x))
```

whereas with `integrate_blindly`, it has the form:

```python
sol = []

DDE.integrate_blindly(t_max_delay, step = 0.01)

for time in DDE.t + np.arange(0,10,0.01):
	sol.append(DDE.integrate(time))
```
 Is this assuming that we are not interested in the behavior between 0.0 and `t_max_delay`? To me it seems that the integration steps for that interval are not accessible, and it's more of an initial adjustment to address the initial discontinuity. Is that correct?

A potential issue is that if you do simply substitute `integrate_blindly` where `integrate` was, i.e.

```python
# Define system...

sol = []
for t_x in T:
    sol.append(DDE.integrate_blindly(t_x))
```
it will overwrite previous values of the `sol` array. I think this is because `sol` is storing references to anchors, and once `jitcdde.DDE.forget` is called, these are discarded. Additionally, looking at the C-layer, `integrate_blindly` calls `static PyObject * get_current_state`, which simply reads the last anchor, while `integrate` calls `static PyObject * get_recent_state`. The latter creates a new object, so you don't get the same issue, even though it is also calling `jitcdde.DDE.forget` during integration. Whether this should be considered an issue, I'm not sure. Because this doesn't seem to be the intended usage from the examples. But let me know your thoughts.

**Comment by LukeLabrie** [comment]
> The integration steps are accessible and returned by `integrate_blindly` for whatever you need. 

`integrate_blindly` returns an array of the computed states at the time given (see below)

https://github.com/neurophysik/jitcdde/blob/27d7177c2f8546fe0d19de7197337ee876f0d931/jitcdde/_jitcdde.py#L910C1-L913C54

So something like 

```python
DDE.integrate_blindly(t_max_delay, step = 0.01)
```
would just give you the state at time `t_max_delay`, not the integration steps. I tried using `DDE.get_state()` to access the integration steps but this too skips straight from the starting time to the time provided to `integrate_blindly`. Perhaps the section below could be modified so that integration steps are stored?

https://github.com/neurophysik/jitcdde/blob/27d7177c2f8546fe0d19de7197337ee876f0d931/jitcdde/_jitcdde.py#L919C1-L927C45

**Comment by mwappner** [comment]
Can you not simply
```python
data = [DDE.integrate_blindly(DDE.t + dt, dt) for _ in range(integration_steps)]
```
and extract the state after each blind integration? I don't think the overhead of the python loop will be too much of an issue, since in the background it's already doing this loop in python, and I'd expect the bottleneck to be the actual integration.

**Comment by Wrzlprmft** [comment]
@mwappner is correct, and I should have expressed this better. So, let me try again:

`integrate_blindly` allows you to access intermediate results the same way as `integrate`. You just need to call it multiple times with the desired step size (assuming that `max_step` and similar do not impose a lower step).

**Comment by LukeLabrie** [comment]
Thanks, this indeed works after the fix discussed above ([12ae330](https://github.com/neurophysik/jitcdde/commit/12ae3302409cab60baa4c0d471a287064f0c5b03)). Any plans for a 1.8.2 release anytime soon?

**Comment by Wrzlprmft** [comment]
There are currently a lot of changes going on thanks to @alexfikl and slowed down by me being exceptionally busy. My intention was to wait for those to be done to some extent before making a release. Ironically, the part I need to do myself is to set up things such that releases are part of the continuous integration and thus easier.

So, if a release right now would make your life easier, please tell me; it is certainly possible.
